planning/planning_test_utils/include/planning_interface_test_manager/planning_interface_test_manager_utils.hpp

```diff
@@ -290,6 +269,127 @@ Scenario makeScenarioMsg(const std::string scenario)
   return scenario_msg;
 }
 
+Pose createPoseFromLaneID(const int & lane_id)
+{
+  auto map_bin_msg = makeMapBinMsg();
+  // create route_handler
+  auto route_handler = std::make_shared<RouteHandler>();
+  route_handler->setMap(map_bin_msg);
+
+  // get middle idx of the lanelet
+  const auto lanelet = route_handler->getLaneletsFromId(lane_id);
+  const auto center_line = lanelet.centerline();
+  const size_t middle_point_idx = std::floor(center_line.size() / 2.0);
+
+  // get middle position of the lanelet
+  geometry_msgs::msg::Point middle_pos;
+  middle_pos.x = center_line[middle_point_idx].x();
+  middle_pos.y = center_line[middle_point_idx].y();
+
+  // get next middle position of the lanelet
+  geometry_msgs::msg::Point next_middle_pos;
+  next_middle_pos.x = center_line[middle_point_idx + 1].x();
+  next_middle_pos.y = center_line[middle_point_idx + 1].y();
+
+  // calculate middle pose
+  geometry_msgs::msg::Pose middle_pose;
+  middle_pose.position = middle_pos;
+  const double yaw = tier4_autoware_utils::calcAzimuthAngle(middle_pos, next_middle_pos);
+  middle_pose.orientation = tier4_autoware_utils::createQuaternionFromYaw(yaw);
+
+  return middle_pose;
+}
+
+Odometry makeInitialPoseFromLaneId(const int & lane_id)
+{
+  Odometry current_odometry;
+  current_odometry.pose.pose = createPoseFromLaneID(lane_id);
+  current_odometry.header.frame_id = "map";
+
+  return current_odometry;
+}
+
+RouteSections combine_consecutive_route_sections(
+  const RouteSections & route_sections1, const RouteSections & route_sections2)
+{
+  RouteSections route_sections;
+  route_sections.reserve(route_sections1.size() + route_sections2.size());
+  if (!route_sections1.empty()) {
+    // remove end route section because it is overlapping with first one in next route_section
+    route_sections.insert(route_sections.end(), route_sections1.begin(), route_sections1.end() - 1);
+  }
+  if (!route_sections2.empty()) {
+    route_sections.insert(route_sections.end(), route_sections2.begin(), route_sections2.end());
+  }
+  return route_sections;
+}
+
+Route::Message makeBehaviorRouteFromLaneId(const int & start_lane_id, const int & goal_lane_id)
```

Comment:
Please write a description for this function. To explain the strategy to make the route.

Reply:
@TakaHoribe 
I added the description I hope it's enough but if not, I'd be happy to add more:smile:

Reply:
Great! Thanks!